addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,794 @@
+ * sent to one of the Tivo's channels.
+ *
+ * @author Jayson Kubilis - Initial contribution
+ * @author Andrew Black (AndyXMB) - Updates / compilation corrections. Addition of channel scanning functionality.
+ */
+
+public class TiVoHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(TiVoHandler.class);
+    private TivoConfigData tivoCfgData = null;
+    private TivoStatusProvider myTivoService = null;
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> chScanJob;
+
+    /**
+     * Instantiates a new TiVo handler.
+     *
+     * @param thing the thing
+     * @param discoveryServiceRegistry the discovery service registry
+     */
+    public TiVoHandler(Thing thing) {
+        super(thing);
+        logger.debug("TiVoHandler '{}' - creating", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        // Handles the commands from the various TiVo channel objects
+        logger.debug("handleCommand {}, command: {}", channelUID, command);
+
+        if (command != null && myTivoService != null) {
+
+            TivoStatusData myTivo = myTivoService.getServiceStatus();
+            String tivoCommand = null;
+
+            if (!isInitialized()) {
+                logger.debug("handleCommand '{}' device is not intialised yet, command '{}' will be ignored.",
+                        getThing().getUID(), channelUID + " " + command);
+                return;
+            }
+
+            // Check to see if we are running a channel scan, if so 'disable' UI commands, else chaos ensues...
+            if (myTivo != null && myTivo.getChScan()) {
+                logger.warn("TiVo '{}' channel scan is in progress, command '{}' will be ignored.", getThing().getUID(),
+                        channelUID + " " + command);
+                return;
+            }
+
+            String tmpAct = command.toString().toUpperCase();
+            if (command instanceof RefreshType) {
+
+                switch (channelUID.getId()) {
+
+                    case CHANNEL_TIVO_STATUS:
+                    case CHANNEL_TIVO_CHANNEL_FORCE:
+                    case CHANNEL_TIVO_CHANNEL_SET:
+                        if (myTivo != null && myTivo.isCmdOk()) {
+                            updateTivoStatus(myTivo);
+                        }
+                        break;
+                    default:
+                        // Future enhancement, if we can come up with a sensible set of actions when a REFRESH is issued
+                        logger.info("TiVo '{}' skipping REFRESH command for channel: '{}'.", getThing().getUID(),
+                                channelUID.getId());
+                }
+
+                return;
+            }
+
+            switch (channelUID.getId())
+
+            {
+                case CHANNEL_TIVO_CHANNEL_FORCE:
+                    tivoCommand = "FORCECH";
```

Comment:
Is this an intended fall-through, they are so rare that they deserve a comment or should not be used, what about just calling a method with either FORCECH or SETCH.

I think it might also be an idea to wrap the real command values behind an enum so you can use proper names in your code

So `TivoCommand.FORCE_CHANNEL` could refer to the value `FORCECH`